lfs_ffi.lua

```diff
@@ -991,51 +1223,56 @@ local attr_handlers = {
     size = function(st) return tonumber(st.st_size) end,
     uid = function(st) return tonumber(st.st_uid) end,
 }
+local mt = {
+    __index = function(self, attr_name)
+        local func = attr_handlers[attr_name]
+        return func and func(self)
+    end
+}
+local stat_type = ffi.metatype('lfs_stat', mt)
 
 -- Add target field for symlinkattributes, which is the absolute path of linked target
 local get_link_target_path
 if OS == 'Windows' then
+    local ENOSYS = 40
     function get_link_target_path()
-        return nil
+        return nil, "could not obtain link target: Function not implemented ",ENOSYS
     end
 else
-    ffi.cdef('unsigned long readlink(const char *path, char *buf, size_t bufsize);')
-    function get_link_target_path(link_path)
-        local size = MAXPATH
-        while true do
-            local buf = ffi.new('char[?]', size)
-            local read = lib.readlink(link_path, buf, size)
-            if read == -1 then
-                return nil, errno()
-            end
-            if read < size then
-                return ffi_str(buf)
-            end
-            size = size * 2
+    if jit.version_num < 20100 then
+        ffi.cdef('typedef int ssize_t;')
```

Comment:
It seems that the type should be `long` (which is 64 bits in 64 bits platform) instead of `int`.

Reply:
Maybe we could apply https://github.com/spacewander/luafilesystem/commit/9c519dcbcd09c0b9972696a39e3562cdc9311877 to this branch.

Reply:
Informations in internet about long type are always confusing for me.

What is clear in LuaJIT codebase is that ssize_t is CTID_INT_PSZ which is CTID_INT64 if LJ_64 and CTID_INT32 elsewhere.

So perhaps definition should be int64_t or int32_t depending on ffi.abi?

Reply:
@sonoro1234 
Thank you for the information. Yes, `intptr_t` is better.